site/src/middleware/damRewrite.ts

```diff
@@ -0,0 +1,12 @@
+import { NextRequest, NextResponse } from "next/server";
+
+import { CustomMiddleware } from "./chain";
+
+export function withDamRewriteMiddleware(middleware: CustomMiddleware) {
```

Comment:
Due to this is a very essential part of the application - every request goes through those middelwares, I would prefer to have some more information / documentation on what the middlewares are doing. Even though this is  function is indeed very simple (the others in my opinion are more advanced) i would prefer to have something like this for the middlewares:

```suggestion
/**
* This function rewrites requests starting with `/dam` to the internal resource location
*/
export function withDamRewriteMiddleware(middleware: CustomMiddleware) {
```

Reply:
I try to avoid comments that give no additional information. Instead I give my best with self explaining function names. 

DomainRewrite really needs explanation, I'll add a comment. DamRewrite imho doesn't.